Tools/cases_generator/generators_common.py

```diff
@@ -9,10 +9,40 @@
     analysis_error,
 )
 from cwriter import CWriter
-from typing import Callable, Mapping, TextIO, Iterator
+from typing import Callable, Mapping, TextIO, Iterator, Iterable
 from lexer import Token
-from stack import Stack
+from stack import Stack, Local, Storage, StackError
 
+# Set this to true for voluminous output showing state of stack and locals
+PRINT_STACKS = False
+
+class TokenIterator:
+
+    look_ahead: Token | None
+    iterator: Iterator[Token]
+
+    def __init__(self, tkns: Iterable[Token]):
+        self.iterator = iter(tkns)
+        self.look_ahead = None
+
+    def __iter__(self) -> "TokenIterator":
+        return self
+
+    def __next__(self) -> Token:
+        if self.look_ahead is None:
+            return next(self.iterator)
+        else:
+            res = self.look_ahead
+            self.look_ahead = None
+            return res
+
+    def peek(self) -> Token | None:
+        if self.look_ahead is None:
+            try:
+                self.look_ahead = next(self.iterator)
+            except StopIteration:
+                pass
```

Comment:
I think this should do the trick.

```suggestion
			self.look_ahead = next(self.iterator, None)
```